common/cache/domainCache.go

```diff
@@ -98,7 +98,7 @@ type (
 		GetCacheSize() (sizeOfCacheByName int64, sizeOfCacheByID int64)
 	}
 
-	domainCache struct {
+	DefaultDomainCache struct {
```

Comment:
why do you export this struct?

Reply:
because that's the right way to do it :) Constructors should return structs not interfaces. Without this I wouldn't be able to expose `WithBlah` options